Patch-Source: https://github.com/WerWolv/ImHex/pull/2766

Deleted some unrelated changes and added extra replacements.

From 2bce6b740f2bc986abcbfcff37ee2cccf61c7114 Mon Sep 17 00:00:00 2001
From: paxcut <paxcut@outlook.com>
Date: Tue, 23 Jun 2026 11:24:55 -0700
Subject: [PATCH] build: archlinux and MSVC

Archlinux needs to change some headers to include fmt/format.hpp instead of fmt/core.fmt. MSVC builds had last change reverted because not only it did not fix the build errors, but it added some extra ones.
---
 .github/workflows/build.yml                           | 2 --
 lib/libimhex/include/hex/api/localization_manager.hpp | 2 +-
 lib/libimhex/include/hex/helpers/logger.hpp           | 2 +-
 3 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/libimhex/include/hex/api/localization_manager.hpp b/lib/libimhex/include/hex/api/localization_manager.hpp
index be8c40b7685ca..ad560d4f2238e 100644
--- a/lib/libimhex/include/hex/api/localization_manager.hpp
+++ b/lib/libimhex/include/hex/api/localization_manager.hpp
@@ -8,7 +8,7 @@
 #include <vector>
 #include <functional>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <wolv/types/static_string.hpp>
 
 EXPORT_MODULE namespace hex {
diff --git a/lib/libimhex/include/hex/helpers/logger.hpp b/lib/libimhex/include/hex/helpers/logger.hpp
index f046ca9d45a2d..b02a328996025 100644
--- a/lib/libimhex/include/hex/helpers/logger.hpp
+++ b/lib/libimhex/include/hex/helpers/logger.hpp
@@ -2,7 +2,7 @@
 
 #include <hex.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <fmt/color.h>
 
 #include <wolv/io/file.hpp>
--- a/lib/external/disassembler/lib/source/spec/opcode.cpp
+++ b/lib/external/disassembler/lib/source/spec/opcode.cpp
@@ -5,7 +5,7 @@
 #include <wolv/utils/string.hpp>
 #include <wolv/math_eval/math_evaluator.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 namespace disasm::spec {
 
--- a/lib/external/pattern_language/lib/include/pl/core/errors/error.hpp
+++ b/lib/external/pattern_language/lib/include/pl/core/errors/error.hpp
@@ -3,7 +3,7 @@
 #include <pl/helpers/types.hpp>
 #include <pl/core/location.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <string>
 #include <utility>
--- a/lib/external/pattern_language/lib/include/pl/core/evaluator.hpp
+++ b/lib/external/pattern_language/lib/include/pl/core/evaluator.hpp
@@ -18,7 +18,7 @@
 #include <pl/core/errors/runtime_errors.hpp>
 #include <pl/core/ast/ast_node_type_appilication.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <fmt/ranges.h>
 #include <pl/pattern_language.hpp>
 
--- a/lib/external/pattern_language/lib/include/pl/core/lexer.hpp
+++ b/lib/external/pattern_language/lib/include/pl/core/lexer.hpp
@@ -5,7 +5,7 @@
 
 #include <pl/core/token.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <optional>
 #include <string>
--- a/lib/external/pattern_language/lib/include/pl/core/parser.hpp
+++ b/lib/external/pattern_language/lib/include/pl/core/parser.hpp
@@ -18,7 +18,7 @@
 
 #include <pl/core/errors/result.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <functional>
 #include <map>
--- a/lib/external/pattern_language/lib/include/pl/patterns/pattern.hpp
+++ b/lib/external/pattern_language/lib/include/pl/patterns/pattern.hpp
@@ -6,7 +6,7 @@
 #include <pl/helpers/types.hpp>
 #include <pl/helpers/utils.hpp>
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <wolv/utils/core.hpp>
 #include <wolv/utils/guards.hpp>
